modules/exploits/windows/smb/ms17_010_eternalblue.rb

```diff
@@ -263,10 +263,20 @@ def verify_arch
       '71710533-beba-4937-8319-b5dbef9ccc36', '1.0'
     ).first
 
-    sock = connect(false,
-      'RHOST' => rhost,
-      'RPORT' => 135
-    )
+    begin
+      sock = connect(false,
+        'RHOST' => rhost,
+        'RPORT' => 135
+      )
+    rescue ::Errno::ECONNRESET,
+           ::Rex::HostUnreachable,
```

Comment:
`::Rex::ConnectionError` encompasses all of these and more. You should rescue that instead of these three.